pkg/splunk/enterprise/configuration_test.go

```diff
@@ -845,3 +852,129 @@ func TestAreRemoteVolumeKeysChanged(t *testing.T) {
+
+	// Define ephemeral for etc & PVCs for var
+	spec = &enterprisev1.CommonSplunkSpec{
+		EtcVolumeStorageConfig: enterprisev1.StorageClassSpec{
+			EphemeralStorage: true,
+		},
+		VarVolumeStorageConfig: enterprisev1.StorageClassSpec{
+			StorageCapacity:  "25Gi",
+			StorageClassName: "gp2",
+		},
+	}
+	test(`{"kind":"StatefulSet","apiVersion":"apps/v1","metadata":{"name":"test-statefulset","namespace":"test","creationTimestamp":null},"spec":{"replicas":1,"selector":null,"template":{"metadata":{"creationTimestamp":null},"spec":{"volumes":[{"name":"mnt-splunk-etc","emptyDir":{}}],"containers":[{"name":"splunk","image":"test","resources":{},"volumeMounts":[{"name":"mnt-splunk-etc","mountPath":"/opt/splunk/etc"},{"name":"pvc-var","mountPath":"/opt/splunk/var"}]}]}},"volumeClaimTemplates":[{"metadata":{"name":"pvc-var","namespace":"test","creationTimestamp":null},"spec":{"accessModes":["ReadWriteOnce"],"resources":{"requests":{"storage":"25Gi"}},"storageClassName":"gp2"},"status":{}}],"serviceName":"","updateStrategy":{}},"status":{"replicas":0}}`)
+
+	// Define ephemeral for etc & var(should ignore storage capacity & storage class name)
+	spec = &enterprisev1.CommonSplunkSpec{
+		EtcVolumeStorageConfig: enterprisev1.StorageClassSpec{
+			EphemeralStorage: true,
+		},
+		VarVolumeStorageConfig: enterprisev1.StorageClassSpec{
+			EphemeralStorage: true,
+			StorageCapacity:  "25Gi",
+			StorageClassName: "gp2",
+		},
+	}
+	test(`{"kind":"StatefulSet","apiVersion":"apps/v1","metadata":{"name":"test-statefulset","namespace":"test","creationTimestamp":null},"spec":{"replicas":1,"selector":null,"template":{"metadata":{"creationTimestamp":null},"spec":{"volumes":[{"name":"mnt-splunk-etc","emptyDir":{}},{"name":"mnt-splunk-var","emptyDir":{}}],"containers":[{"name":"splunk","image":"test","resources":{},"volumeMounts":[{"name":"mnt-splunk-etc","mountPath":"/opt/splunk/etc"},{"name":"mnt-splunk-var","mountPath":"/opt/splunk/var"}]}]}},"serviceName":"","updateStrategy":{}},"status":{"replicas":0}}`)
+
+	// Define invalid EtcVolumeStorageConfig
+	spec = &enterprisev1.CommonSplunkSpec{
+		EtcVolumeStorageConfig: enterprisev1.StorageClassSpec{
+			StorageCapacity: "----",
+		},
+	}
+	err := addStorageVolumes(&cr, spec, statefulSet, labels)
+	if err == nil {
+		t.Errorf("Unable to idenitfy incorrect EtcVolumeStorageConfig resource quantity")
+	}
+
+	// Define invalid VarVolumeStorageConfig
+	spec = &enterprisev1.CommonSplunkSpec{
+		VarVolumeStorageConfig: enterprisev1.StorageClassSpec{
+			StorageCapacity: "----",
+		},
+	}
+	err = addStorageVolumes(&cr, spec, statefulSet, labels)
+	if err == nil {
+		t.Errorf("Unable to idenitfy incorrect VarVolumeStorageConfig resource quantity")
+	}
+
```

Comment:
It _may_ be good to have `test` here to confirm the previous values have not been wrongly/mistakenly overwritten when invalid configs are applied.

Reply:
For now, this test was only for code coverage purposes of an error leg. The erroneous config test might not help here. Good idea though, will keep it in mind for the future though.